### PR TITLE
CAMEL-19991: Jetty and SB upgrades

### PR DESCRIPTION
**Pull request by Croway**
Upgrade to jetty 12 and spring-boot 3.2.0 alignments

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: components/camel-cxf/camel-cxf-rest/src/main/java/org/apache/camel/component/cxf/jaxrs/CxfRsProducer.java [hunk 1]
```diff
@@ -24,11 +24,7 @@
 import java.net.CookieStore;
 import java.net.HttpCookie;
 import java.net.URLDecoder;
-import java.util.Collection;
-import java.util.HashMap;
-import java.util.LinkedHashMap;
-import java.util.List;
-import java.util.Map;
+import java.util.*;
```
Comment (essobedo) [inline components/camel-cxf/camel-cxf-rest/src/main/java/org/apache/camel/component/cxf/jaxrs/CxfRsProducer.java:27]: Please avoid wildcard import

Comment (Croway) [reply components/camel-cxf/camel-cxf-rest/src/main/java/org/apache/camel/component/cxf/jaxrs/CxfRsProducer.java:27]: yeah, the IDE thinks he's smart, I'll review them and update the PR

**Review by essobedo — COMMENTED** [review]
Very nice work 👏 Just the comments/concerns. One of my concerns is about the replacement of Jetty by undertow in CXF, it sounds like very risky

File: components/camel-jetty/src/test/java/org/apache/camel/component/jetty/MultiPartFormTest.java [hunk 2]
```diff
@@ -98,7 +98,8 @@ public void process(Exchange exchange) throws Exception {
                         // "text/plain", data.getContentType());
                         assertEquals("log4j2.properties", data.getName(), "Got the wrong name");
 
-                        assertTrue(data.getDataSource().getInputStream().available() > 0,
+                        String fileContent = new String(data.getDataSource().getInputStream().readAllBytes());
+                        assertTrue(fileContent.length() > 0,
```
[the remark this answers is not in the document]
Comment (Croway) [reply components/camel-jetty/src/test/java/org/apache/camel/component/jetty/MultiPartFormTest.java:102]: oh yeah, nice, this way the whole stream won't loaded in memory, thanks

File: components/camel-jetty/src/main/java/org/apache/camel/component/jetty11/CamelInputStreamContentProvider.java [hunk 3]
```diff
@@ -18,9 +18,9 @@
 
 import java.io.InputStream;
 
-import org.eclipse.jetty.client.util.InputStreamContentProvider;
+import org.eclipse.jetty.io.content.InputStreamContentSource;
 
-public class CamelInputStreamContentProvider extends InputStreamContentProvider {
+public class CamelInputStreamContentProvider extends InputStreamContentSource {
```
[the remark this answers is not in the document]
Comment (davsclaus) [reply components/camel-jetty/src/main/java/org/apache/camel/component/jetty11/CamelInputStreamContentProvider.java:23]: Yes it makes sense to rename package from 11 to 12 wherever

File: components/camel-openstack/src/test/java/org/apache/camel/component/openstack/it/OpenstackWiremockTestSupport.java [hunk 4]
```diff
@@ -32,7 +31,6 @@ public class OpenstackWiremockTestSupport extends CamelTestSupport {
     @BeforeAll
     public static void startServer() {
         WireMockConfiguration configuration = options().dynamicPort();
-        configuration.extensions(new ResponseTemplateTransformer(false));
```
[the remark this answers is not in the document]
Comment (Croway) [reply components/camel-openstack/src/test/java/org/apache/camel/component/openstack/it/OpenstackWiremockTestSupport.java:35]: WireMock is not clear about it, I could not find any migration/upgrade guide or discussion about it on github

**Comment by davsclaus** [comment]
okay there are some test faulures that is worth to look at first and also the merge conflict now


rg.apache.camel.component.http.HttpProducerSessionTest.testExchangeSession | 180.4s | ERROR
-- | -- | --
org.apache.camel.component.http.HttpProducerSessionTest.testNoSession | 180.2s | ERROR
org.apache.camel.component.http.HttpProducerSessionTest.testInstanceSession | 180.3s | ERROR
org.apache.camel.component.http.HttpNoConnectionRedeliveryTest.httpConnectionNotOk | 180.1s | FAILURE
org.apache.camel.component.salesforce.RawPayloadTest.testRestApi | 0.687s | ERROR
org.apache.camel.component.salesforce.RawPayloadTest.testRestApi | 0.119s | ERROR
org.apache.camel.component.salesforce.RawPayloadTest.testRestApi | 0.104s | ERROR
org.apache.camel.component.salesforce.RawPayloadTest.testRestApi | 0.098s | ERROR
org.apache.camel.component.salesforce.RawPayloadTest.testRestApi | 0.094s | ERROR
org.apache.camel.component.salesforce.RawPayloadTest.testRestApi | 0.084s | ERROR
org.apache.camel.component.salesforce.RawPayloadTest.testRestApi | 0.093s | ERROR
org.apache.camel.component.salesforce.RawPayloadTest.testRestApi | 0.089s | ERROR
org.apache.camel.component.salesforce.RawPayloadTest.testRestApi | 0.086s | ERROR
org.apache.camel.component.salesforce.RawPayloadTest.testRestApi | 0.097s | ERROR
org.apache.camel.component.salesforce.RawPayloadTest.testRestApi | 0.085s | ERROR
org.apache.camel.component.salesforce.RawPayloadTest.testRestApi | 0.082s | ERROR
org.apache.camel.component.salesforce.RawPayloadTest.testRestApi | 0.089s | ERROR
org.apache.camel.component.salesforce.RawPayloadTest.testRestApi | 0.077s | ERROR
org.apache.camel.component.salesforce.RawPayloadTest.testRestApi | 0.082s | ERROR
org.apache.camel.component.salesforce.RawPayloadTest.testRestApi | 0.090s | ERROR
org.apache.camel.component.salesforce.RawPayloadTest.testRestApi | 0.076s | ERROR
org.apache.camel.component.salesforce.RawPayloadTest.testRestApi | 0.078s | ERROR
org.apache.camel.component.salesforce.RawPayloadTest.testRestApi | 0.082s | ERROR
org.apache.camel.component.salesforce.RawPayloadTest.testRestApi | 0.081s | ERROR
org.apache.camel.component.salesforce.RawPayloadTest.testRestApi | 0.078s | ERROR
org.apache.camel.component.salesforce.RawPayloadTest.testRestApi | 0.079s | ERROR
org.apache.camel.component.salesforce.RawPayloadTest.testRestApi | 0.082s | ERROR
org.apache.camel.component.salesforce.RawPayloadTest.testRestApi | 0.077s | ERROR
org.apache.camel.component.salesforce.RawPayloadTest.testRestApi | 0.075s | ERROR
org.apache.camel.component.salesforce.RawPayloadTest.testRestApi | 0.073s | ERROR
org.apache.camel.component.salesforce.RawPayloadTest.testRestApi | 0.090s | ERROR
org.apache.camel.component.salesforce.RawPayloadTest.testRestApi | 0.079s | ERROR
org.apache.camel.component.salesforce.RawPayloadTest.testRestApi | 0.074s | ERROR
org.apache.camel.component.salesforce.RawPayloadTest.testRestApi | 0.074s | ERROR
org.apache.camel.component.salesforce.RawPayloadTest.testRestApi | 0.066s | ERROR
org.apache.camel.component.salesforce.RawPayloadTest.testRestApi | 0.070s | ERROR
org.apache.camel.component.salesforce.RawPayloadTest.testRestApi | 0.070s | ERROR
org.apache.camel.component.salesforce.RawPayloadTest.testRestApi | 0.066s | ERROR
org.apache.camel.component.salesforce.RawPayloadTest.testRestApi | 0.074s | ERROR
org.apache.camel.component.salesforce.RawPayloadTest.testRestApi | 0.074s | ERROR
org.apache.camel.component.salesforce.RawPayloadTest.testRestApi | 0.070s | ERROR
org.apache.camel.component.salesforce.RawPayloadTest.testRestApi | 0.066s | ERROR
org.apache.camel.component.salesforce.RawPayloadTest.testRestApi | 0.074s | ERROR
org.apache.camel.component.salesforce.PubSubApiTest.testReconnectOnErrorAfterReplayIdNonNull | 3.055s | FAILURE

**Review by ffang — APPROVED** [review]

**Comment by davsclaus** [comment]
org.apache.camel.component.jetty.MultiPartFormBigFileTest.testSendMultiPartFormWithBigFile | 10.39s | FAILURE
-- | -- | --

**Comment by Croway** [comment]
> org.apache.camel.component.jetty.MultiPartFormBigFileTest.testSendMultiPartFormWithBigFile 	10.39s 	FAILURE

On it already, seems like that jetty 12 does not flush the multipart file in the tmp folder

**Comment by Croway** [comment]
@davsclaus finally, tests are green

**Comment by davsclaus** [comment]
Awesome, so lets merge this

**Pull request merged**